apps/remix-ide/src/app/plugins/remixAIPlugin.tsx

```diff
@@ -1,43 +1,67 @@
 import * as packageJson from '../../../../../package.json'
 import { ViewPlugin } from '@remixproject/engine-web'
 import { Plugin } from '@remixproject/engine';
-import { RemixAITab } from '@remix-ui/remix-ai'
-import React from 'react';
-import { ICompletions, IModel, RemoteInferencer, IRemoteModel } from '@remix/remix-ai-core';
+import { RemixAITab, ChatApi } from '@remix-ui/remix-ai'
+import React, { useCallback } from 'react';
+import { ICompletions, IModel, RemoteInferencer, IRemoteModel, IParams, GenerationParams, CodeExplainAgent } from '@remix/remix-ai-core';
+
+type chatRequestBufferT<T> = {
+  [key in keyof T]: T[key]
+}
 
 const profile = {
   name: 'remixAI',
   displayName: 'Remix AI',
   methods: ['code_generation', 'code_completion',
     "solidity_answer", "code_explaining",
     "code_insertion", "error_explaining",
-    "initialize"],
+    "initialize", 'chatPipe', 'ProcessChatRequestBuffer', 'isChatRequestPending'],
   events: [],
   icon: 'assets/img/remix-logo-blue.png',
   description: 'RemixAI provides AI services to Remix IDE.',
   kind: '',
-  // location: 'sidePanel',
+  location: 'sidePanel',
   documentation: 'https://remix-ide.readthedocs.io/en/latest/remixai.html',
   version: packageJson.version,
   maintainedBy: 'Remix'
 }
 
-export class RemixAIPlugin extends Plugin {
+export class RemixAIPlugin extends ViewPlugin {
   isOnDesktop:boolean = false
   aiIsActivated:boolean = false
   readonly remixDesktopPluginName = 'remixAID'
   remoteInferencer:RemoteInferencer = null
   isInferencing: boolean = false
+  chatRequestBuffer: chatRequestBufferT<any> = null
+  agent: CodeExplainAgent
 
   constructor(inDesktop:boolean) {
     super(profile)
     this.isOnDesktop = inDesktop
-
+    this.agent = new CodeExplainAgent(this)
     // user machine dont use ressource for remote inferencing
   }
 
   onActivation(): void {
-    this.initialize(null, null, null, false)
+    if (this.isOnDesktop) {
+      console.log('Activating RemixAIPlugin on desktop')
+      this.on(this.remixDesktopPluginName, 'activated', () => {
+        this.call("remixAI", 'initialize', null, null, null, false);
+      })
+    } else {
+      console.log('Activating RemixAIPlugin on browser')
+      this.initialize()
+    }
+    this.setRemixAIOnSidePannel(false)
+  }
+
+  setRemixAIOnSidePannel(resize:boolean=false){
+    if (resize){
+      this.call('sidePanel', 'pinView', profile)
```

Comment:
`if` and `else` are doing the same thing.

Reply:
Waiting for some answers from David to resize the sidepanel

Reply:
we will remove it from there anyway, you don't need to do this @STetsing